dot/network/service.go

```diff
@@ -521,6 +535,9 @@ func (s *Service) readStream(stream libp2pnetwork.Stream, peer peer.ID, decoder
 			_ = stream.Close()
 			return
 		}
+
+		// todo (ed) determine if there are other places to capture data received
+		s.host.bwc.LogRecvMessage(int64(tot))
```

Comment:
all messages received by the network go through this function, and no where else, so having this just here is fine

Reply:
Excellent, thanks.